addons/mail/models/mail_channel.py

```diff
@@ -592,6 +592,7 @@ def channel_info(self, extra_info=False):
                     info['is_minimized'] = partner_channel.is_minimized
                     info['seen_message_id'] = partner_channel.seen_message_id.id
                     info['custom_channel_name'] = partner_channel.custom_channel_name
+                    info['is_pinned'] = partner_channel.is_pinned
```

Comment:
This may prove problematic.
In action_unfollow, we build channel_info before removing the relation, which, when arriving in the js just won't hold the right information

Reply:
Thanks for the heads up, that method is dumb...
Quick fix would be to overwrite `is_pinned` in the result there.
But any other similar side effect would go unnoticed...

Ideally method should not return channel info is channel becomes inaccessible for current user, it should return "delete channel" then.